### ISSUE TITLE
Filtering & paging for the link list

### ISSUE THREAD

**Issue by Jako** [tinymce/tinymce#6588]
It seems that the link listbox does not provide any paging or filtering for the list. Is this somehow possible?

Would it be possible to add something the single 'Single select input' demo on https://joshuajohnson.co.uk/Choices/ to the link list field?

**Comment by lostkeys** [comment]
Yeah this is not something we have today. I'll flag this as a feature request. Just to get an idea of the use case, how many links do you have your end users pick from, roughly?

**Comment by Jako** [comment]
The user has ~200k resources in the CMS. So the list contains ~200k entries. This needs some filtering to produce an empty list by a connector option.

**Comment by lostkeys** [comment]
I imagined you would give me that answer 😊. We have some ideas on an improved UI for this, but we do not have any concrete plans as of today. Personally it sort of surprises me that TinyMCE doesn't already has this given it's history in the CMS industry. Being able to browse and search site pages does seem like a standard CMS feature.

**Comment by Jako** [comment]
It needs either a tree loading connector that loads just one level of the tree or a list loading connector having limit/offset/search properties. The second one should be easier to implement.

**Comment by shubhangigavhane8** [comment]
[bnvnvn](url)

**Comment by Jako** [comment]
Are there any new plans in this regard?

**Comment by Gallenkamp** [comment]
I, too, would like this search box back. A dropdown is just not what you need when working with more than 3 documents.
Is there any way to speed this up?

**Comment by Jako** [comment]
@Gallenkamp The search box was a custom addition to the MODX TinyMCE package. It was never part of TinyMCE.

The current component code of TinyMCE lacks a searchable combo box and an external script like choices, select2 etc. can't be used that easy as in TinyMCE 4.

**Comment by Anu3ev** [comment]
Hello! Is there any new information here? In the image plugin, I have an image list with 300 links. It is not  It's inconvenient to scroll through such a list with links to pictures.

I would be grateful if I could turn on some setting option that allowed me to use the search input with the listbox, not just the listbox.

**Comment by Jako** [comment]
Are there any further plans on this?

**Comment by MariamikGhazaryan** [comment]
Is there any news about this issue?? Will it be fixed soon? I also have dropdown with a lot of options.

**Comment by pxim** [comment]
Where is the source code of this listbox? Can it be customized and modified and then introduced and used?

![Image](https://github.com/user-attachments/assets/e454ae78-e1a0-4c58-a240-b617b7fa07ff)
